File: rando/TaylorSeries_SquareRoot.c
#include <stdio.h>
#include <math.h>

int main() {
    while (1) {    
        float num;
        printf("Enter a number to find its square root: ");
        scanf("%f", &num);
        if (num < 0) {
            printf("Negative numbers do not have real square roots.\n");
            return 0;
        }
        float sqrt_approx;
        if (num < 1) {
            sqrt_approx = 1 + (num - 1)/2 - pow(num - 1, 2)/8 + pow(num - 1, 3)/16 - 5*pow(num - 1, 4)/128;
            printf("The square root of %f is approximately %f\n", num, sqrt_approx);
        } else {
            int a = 1;
            while (1) {
                if (a * a < num) {
                    a += 1;
                    printf("a = %d, a ^ 2 = %d --> %f\n", a, a * a, num);
                } else {
                    break;
                }
            }
            a -= 1;
            float x = num / (a * a);
            sqrt_approx = a + (x) / (2 * a) + pow(x, 2) / (8 * pow(a, 3));
        }
        printf("The square root of %f is approximately %f\n", num, sqrt_approx);
        printf("The actual square root of %f is %f\n", num, sqrt(num));
    }
    return 0;
}

File: Training_Task_1/rando/TaylorSeries_SquareRoot.py
from math import factorial


num = input("Enter a number to find its square root: ")
num = float(num)
if num < 1: 
    sqrt_approx = 1 + (num - 1)/2 - (num - 1)**2/8 + (num - 1)**3/16 - 5*(num - 1)**4/128
else:
    a = 1
    while True:
        if a ** 2 < num:
            a += 1
            print(f'a = {a}, a^2 = {a ** 2} ---> {num}')
        else:
            break
    a -= 1
    x = num / (a ** 2)
    sqrt_approx = a + (x) / (2 * a) + (x) ** 2 / (8 * a ** 3)







print(f"Approximate square root of {num} using Taylorseries: {sqrt_approx}")
# Calculate the actual square root for comparison
actual_sqrt = num ** 0.5
print(f"Actual square root of {num}: {actual_sqrt}")

